src/TSTransformer/util/offset.ts

```diff
@@ -16,6 +16,20 @@ export function offset(expression: luau.Expression, value: number) {
 	if (value === 0) {
 		return expression;
 	}
+	// special case to handle adding and removing the offset
+	if (expression.kind === luau.SyntaxKind.BinaryExpression) {
+		if (expression.right.kind === luau.SyntaxKind.NumberLiteral) {
+			if (expression.operator === "+" || expression.operator === "-") {
+				const rightValue = getLiteralNumberValue(expression.right)!;
```

Comment:
We should add an overload that guarantees this return type when a `luau.NumberLiteral` is passed, instead of having to ugly non-null assert.